lib/Runtime/PlatformAgnostic/ICU.h

```diff
@@ -8,7 +8,15 @@
 
 #ifdef HAS_ICU
 #ifdef WINDOWS10_ICU
+// if WINDOWS10_ICU is defined, pretend like we are building for recent Redstone,
+// even if that isn't necessarily true
+#ifdef NTDDI_VERSION
+#undef NTDDI_VERSION
+#endif
+#define NTDDI_VERSION NTDDI_WIN10_RS3
 #include <icu.h>
+#include <icucommon.h>
+#include <icui18n.h>
```

Comment:
Win10 ICU actually requires all 3 headers?

Reply:
It wasn't building without the extra two, and I think the reason why after looking into it a bit more is that it thinks <icu.h> is the current file, "ICU.h". Which is silly. Not sure how to work around that but I can probably just remove the include for icu.h (how is a self-referential include not horribly broken????)

Reply:
Well angle brackets (as distinct from quotes) are supposed to skip the local path lookup logic and go directly to the configured include paths and then system paths, so `<icu.h>` and `"icu.h"` **should** be different, but it's possible the way our include paths are set up includes this directory anyway. The fix would be to not do that for this compilation unit, or rename this file. Honestly `ChakraICU.h` is probably a better naming choice to avoid confusion. (Although that might get confusing with the Chakra.ICU.*.dll targets?)

Reply:
That was my original rationale for not naming it ChakraICU.h. 

https://msdn.microsoft.com/en-us/library/36k2cdd4.aspx?f=255&MSPPError=-2147217396 suggests that <> still looks in /I paths, and I believe most of our projects are set up to specifically include MSBuildThisFileDirectory in <AdditionalIncludes>